### ISSUE TITLE
paddleocr 使用e2e模型训练 CCPD车牌数据 nanxxx问题？

### ISSUE THREAD

**Issue by xuwuwei** [PaddlePaddle/PaddleOCR#3011]
**这是日志**
[2021/06/03 00:17:25] root INFO: train from scratch
[2021/06/03 00:17:25] root INFO: train dataloader has 1000 iters
[2021/06/03 00:17:25] root INFO: valid dataloader has 5000 iters
[2021/06/03 00:17:25] root INFO: During the training process, after the 0th iteration, an evaluation is run every 1000 iterations
[2021/06/03 00:17:25] root INFO: Initialize indexs of datasets:['D:/ai/PaddleOCR-release-2.1/datasets/license/train.txt']
[2021/06/03 00:17:35] root INFO: epoch: [1/600], iter: 10, lr: 0.000100, loss: nanxxx, score_loss: 0.989663, border_loss: 0.410378, direction_loss: nanxxx, ctc_loss: nanxxx, reader_cost: 0.05210 s, batch_cost: 0.84410 s, samples: 55, ips: 6.51583
[2021/06/03 00:17:43] root INFO: epoch: [1/600], iter: 20, lr: 0.000100, loss: nanxxx, score_loss: 0.989537, border_loss: nanxxx, direction_loss: nanxxx, ctc_loss: nanxxx, reader_cost: 0.00080 s, batch_cost: 0.62698 s, samples: 50, ips: 7.97467

**这是数据**
00205459770115-90_85-352&516_448&547-444&547_368&549_364&517_440&515-0_0_22_10_26_29_24-128-7.jpg	[{"transcription": "AYL250", "points": [[364.0, 515.0], [444.0, 515.0], [444.0, 549.0], [364.0, 549.0]]},{"transcription": "###","points": [[364.0, 481.0], [444.0, 481.0], [444.0, 515.0], [364.0, 515.0]]}]
00221264367816-91_91-283&519_381&553-375&551_280&552_285&514_380&513-0_0_7_26_17_33_29-95-9.jpg	[{"transcription": "AH2T95", "points": [[280.0, 513.0], [380.0, 513.0], [380.0, 552.0], [280.0, 552.0]]},{"transcription": "###","points": [[280.0, 474.0], [380.0, 474.0], [380.0, 513.0], [280.0, 513.0]]}]
00223060344828-90_89-441&517_538&546-530&552_447&548_447&512_530&516-0_0_13_16_33_30_33-148-14.jpg	[{"transcription": "APS969", "points": [[447.0, 512.0], [530.0, 512.0], [530.0, 552.0], [447.0, 552.0]]},{"transcription": "###","points": [[447.0, 472.0], [530.0, 472.0], [530.0, 512.0], [447.0, 512.0]]}]
00224137931034-90_87-351&564_451&606-440&599_362&600_359&572_437&571-0_0_3_21_30_28_24-88-5.jpg	[{"transcription": "ADX640", "points": [[359.0, 571.0], [440.0, 571.0], [440.0, 600.0], [359.0, 600.0]]},{"transcription": "###","points": [[359.0, 542.0], [440.0, 542.0], [440.0, 571.0], [359.0, 571.0]]}]
00227490421455-90_88-341&443_436&482-437&479_338&486_335&452_434&445-0_0_9_33_33_29_28-128-12.jpg	[{"transcription": "AK9954", "points": [[335.0, 445.0], [437.0, 445.0], [437.0, 486.0], [335.0, 486.0]]},{"transcription": "###","points": [[335.0, 404.0], [437.0, 404.0], [437.0, 445.0], [335.0, 445.0]]}]
00229166666667-90_87-362&438_461&467-458&465_364&463_363&432_457&434-0_0_33_29_30_29_28-164-11.jpg	[{"transcription": "A95654", "points": [[363.0, 432.0], [458.0, 432.0], [458.0, 465.0], [363.0, 465.0]]},{"transcription": "###","points": [[363.0, 399.0], [458.0, 399.0], [458.0, 432.0], [363.0, 432.0]]}]

**这是配置**
Global:
  use_gpu: True
  epoch_num: 600
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/pgnet_r50_vd_totaltext/
  save_epoch_step: 10
  # evaluation is run every 0 iterationss after the 1000th iteration
  eval_batch_step: [ 0, 1000 ]
  cal_metric_during_train: False
  pretrained_model:
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img:
  valid_set: totaltext # two mode: totaltext valid curved words, partvgg valid non-curved words
  save_res_path: ./output/pgnet_r50_vd_totaltext/predicts_pgnet.txt
  character_dict_path: ppocr/utils/license.txt
  character_type: EN
  max_text_length: 6 # the max length in seq
  max_text_nums: 2 # the max seq nums in a pic
  tcl_len: 64

Architecture:
  model_type: e2e
  algorithm: PGNet
  Transform:
  Backbone:
    name: ResNet
    layers: 50
  Neck:
    name: PGFPN
  Head:
    name: PGHead

Loss:
  name: PGLoss
  tcl_bs: 64
  max_text_length: 6 # the same as Global: max_text_length
  max_text_nums: 2 # the same as Global：max_text_nums
  pad_num: 6 # the length of dict for pad

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.0001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: PGPostProcess
  score_thresh: 0.5
  mode: fast   # fast or slow two ways

Metric:
  name: E2EMetric
  mode: A   # two ways for eval, A: label from txt,  B: label from gt_mat
  gt_mat_dir:  ./train_data/total_text/gt  # the dir of gt_mat
  character_dict_path: ppocr/utils/license.txt
  main_indicator: f_score_e2e

Train:
  dataset:
    name: PGDataSet
    data_dir: D:\datasets\CCPD2019\ccpd_base
    label_file_list: [D:/ai/PaddleOCR-release-2.1/datasets/license/train.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - E2ELabelEncodeTrain:
      - PGProcessTrain:
          batch_size: 5  # same as loader: batch_size_per_card
          min_crop_size: 24
          min_text_size: 4
          max_text_size: 64
      - KeepKeys:
          keep_keys: [ 'images', 'tcl_maps', 'tcl_label_maps', 'border_maps','direction_maps', 'training_masks', 'label_list', 'pos_list', 'pos_mask' ] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: True
    batch_size_per_card: 5
    num_workers: 10

Eval:
  dataset:
    name: PGDataSet
    data_dir: D:\datasets\CCPD2019\ccpd_base
    label_file_list: [D:/ai/PaddleOCR-release-2.1/datasets/license/train.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: RGB
          channel_first: False
      - E2ELabelEncodeTest:
      - E2EResizeForTest:
          max_side_len: 768
      - NormalizeImage:
          scale: 1./255.
          mean: [ 0.485, 0.456, 0.406 ]
          std: [ 0.229, 0.224, 0.225 ]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: [ 'image', 'shape', 'polys', 'texts', 'ignore_tags', 'img_id']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2

**启动命令**
python tools/train.py -c configs/e2e/e2e_r50_vd_pg_license.yml

**Comment by LDOUBLEV** [comment]
可以检查下direction_loss  ctc_loss 计算的输入是不是对的

**Comment by Evezerest** [comment]
The issue will be closed, if it is not solved or there is a follow-up one, please reopen it at any time and we will continue to follow up. It is recommended to pull and try the latest code first.
该问题将会被关闭，若问题未解决或有后续问题，请随时重新打开（建议先拉取最新代码进行尝试），我们会继续跟进。

**Comment by XinDing5** [comment]
请问您用CCPD训练PGNet后来成功了吗？我也在尝试这个工作，目前的情况是训练的loss很低，但是在训练集和测试集的预测结果都很差，不太清楚是什么原因
